### DISCUSSION TITLE
MQTT on Alpine Linux keeps crashing: anyone here who has done it or know his way around?

### DISCUSSION THREAD

**Discussion by Nick4-1** [Koenkk/zigbee2mqtt#21403] [General]
My knowledge about Linux is basic but I use Debian for quite some time.
Most of the time I get things done because of the many info out there.
I'm running Proxmox with mostly LXC's, and run HA & MQTT broker + zigbee2MQTT in separate containers.

A little while ago I discovered Alpine Linux and since I run on the limit with RAM I have succeeded to migrate a few containers to Alpine.
Now I was trying to do the same for the MQTT + zigbee2MQTT machine but am stuck with MQTT.
Even though there is a package for MQTT on Alpine, something goes wrong and the service doesn't start, has the crashed status.
I have searched for hours and hours but cannot find a reason/solution, probably because of my lack of knowledge.

Is there anyone here who can point me in the right direction?
Any help is highly appreciated!

**Comment by morrisb74** [comment]
Is it crashing after a while, or crashing immediatly after startup ?
If it's crashing after a while, you might want to use Monit to monitor and automatically restart it.
I run it on Debian, but even on Debian I run Monit to keep my Z2M up.

**Reply by Nick4-1** [reply]
Hi @morrisb74, thanks for jumping in!
I had no problem getting it running on Debian but with Alpine, something is different.

Again, for hours I have been searching and trying a lot of things, it seems that I have come closer to the issue:
1) permissions: when changing them (they are quite restricted) I can start the binary manually, and I'm able to connect with HA
2) somehow after rebooting, these things are being reset so the service doesn't start/crashes

EDIT: I think I've nailed it down
The mosquitto.pid file cannot be written due to the directory not existing.
I created it with
`sudo mkdir /var/run/mosquitto/ && sudo chown mosquitto: /var/run/mosquitto`
when I restart the mosquitto service, this works.
After rebooting though, the pid file cannot be written because the '/var/run/mosquitto/' folder is being deleted every time.

**Comment by morrisb74** [comment]
Any specific errors ?
You might want to drop the pid file in /tmp so it's deleted after a reboot but /tmp is always available
